drivers/read_fs.c

```diff
@@ -508,4 +506,26 @@ void __rt_fs_cluster_read(rt_file_t *file, void *buffer, size_t size, rt_fs_req_
   __rt_cluster_push_fc_event(&req->event);
 }
 
+void __rt_fs_cluster_seek_req(void *_req)
+{
+  rt_fs_req_t *req = (rt_fs_req_t *)_req;
+  rt_file_t *file = req->file;
+  unsigned int offset = req->offset;
+  rt_event_t *event = &req->event;
+  __rt_init_event(event, event->sched, __rt_fs_cluster_req_done, (void *)req);
```

Comment:
It seems you modified rt_fs_seek api due to this code. Rt_fs_seek is actually synchronous so you can just call it and right after call __rt_fs_cluster_req_done as a function call without using events